Modules/parsermodule.c

```diff
@@ -666,6 +662,12 @@ validate_node(node *tree)
     for (pos = 0; pos < nch; ++pos) {
         node *ch = CHILD(tree, pos);
         int ch_type = TYPE(ch);
+        if ((ch_type >= NT_OFFSET + _PyParser_Grammar.g_ndfas)
```

Comment:
What if `ch_type` is negative?

This guards `ch_type`, but what about `next_type`?

Reply:
> What if `ch_type` is negative?

Good point! Now fixed.

> This guards `ch_type`, but what about `next_type`?

`next_type` is taken from `_PyParser_Grammar`, which is hardcoded in graminit.c; so, it's guaranteed to be valid.